### ISSUE TITLE
Unable to restore packages on Ubuntu 18.04

### ISSUE THREAD

**Issue by enricoreich** [dotnet/cli#11560]
Hello,

I'm trying to get the latest 3.0 preview7 working on my Ubuntu 18.04 remote server using the latest tar.gz from [core-setup#daily-builds](https://github.com/dotnet/core-setup#daily-builds) for Linux x64.
The dotnet command is working but if I try to install new templates eg. for blazor, the restore process fires up but then -> nothing. The process stays in my process list:

![Screenshot_1](https://user-images.githubusercontent.com/316637/59516876-940a1880-8ec2-11e9-968e-b2e1a6146f65.png)

After running the command, that's all that happens:
![boring](https://user-images.githubusercontent.com/316637/59518633-4d1e2200-8ec6-11e9-80b4-7535689a3f57.gif)

Any ideas? Do I miss something here?
Thanks in advance!

## Steps to reproduce
1. Download latest preview7 tar.gz from [core-setup#daily-builds](https://github.com/dotnet/core-setup#daily-builds) for Linux x64
2. unpack with `tar zxvf dotnet-sdk-latest-linux-x64.tar.gz -C ./dotnet3` (create destination folder befor extracting)
3. setup environment variables `export DOTNET_ROOT=$HOME/dotnet3` & `export PATH=$PATH:$HOME/dotnet3`, I'm using zsh so they goes into my .zshrc (or .zprofile)
4. reopen shell
5. run `dotnet new -i "Microsoft.AspNetCore.Blazor.Templates::3.0.0-preview7.*"`
6. restore process hangs with no output or any message

Also with preview6 bits as root or user, does not make any difference.
It works on Windows 10 and WSL (Ubuntu 18.04) perfectly!

## Expected  behavior
The command should restore necessary files from Nuget and install new templates so they can be used with `dotnet new`

## Actual behavior
Restore process hangs with no output or any message.

## Environment data
`dotnet --info` output:
```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview7-012398
 Commit:    028dd7635a

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /home/---/dotnet3/sdk/3.0.100-preview7-012398/

Host (useful for support):
  Version: 3.0.0-preview7-27813-06
  Commit:  5cfc2cab25

.NET Core SDKs installed:
  3.0.100-preview7-012398 [/home/---/dotnet3/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0-preview7.19311.5 [/home/---/dotnet3/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0-preview7-27813-06 [/home/---/dotnet3/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by livarcocc** [comment]
Does restore of regular packages work? Is this an issue only when installing templates?

**Comment by enricoreich** [comment]
This happens even if I want to create a new project with `dotnet new webapp -n AspWebApp` or run `dotnet restore` on an existing project (created on Windows and copied over).

**Comment by peterhuene** [comment]
Do you get any additional output with `dotnet build -v:diag`?

**Comment by enricoreich** [comment]
@peterhuene Yes, but the process does not exit normally..it hangs and some threads are using 1-3% cpu.

[diag.txt](https://github.com/dotnet/cli/files/3291594/diag.txt)

**Comment by livarcocc** [comment]
@nkolev92 @rrelyea

**Comment by enricoreich** [comment]
I have installed dotnet core sdk 2.2.300 as described in the [installation guide](https://dotnet.microsoft.com/download/linux-package-manager/ubuntu18-04/sdk-2.2.300) and everything works like intended.

**Comment by enricoreich** [comment]
Today I tried build 3.0.100-preview7-012421 and noticed that the there is no `NuGetFallbackFolder` at `/opt/dotnet/sdk` (there I unpacked the tar.gz) and it is not created when I try to install (any) `dotnet new` templates.

**Comment by enricoreich** [comment]
I installed the sdk with the help of the dotnet-install.sh and and following command: `./dotnet-install.sh -Channel 3.0 --verbose`.
It downloads and installs 3.0.100-preview4-010988 and everything seems to work correctly. I can install templates, build projects and restore packages.

Using Channel master `./dotnet-install.sh -Channel master --verbose` gets me 3.0.100-preview7-012424 and all the issues that I have described appear again.

**Comment by nkolev92** [comment]
From the logs it seems stuck in the msbuild part of restore, not the restore task itself. 
I no context on similar issues.

**Comment by enricoreich** [comment]
Thanks @nkolev92 for looking into it.

I duno if this helps but I have created diag logs for preview4 (which is working) and preview7 (latest master, not working!) by creating a simple `dotnet new console` ConsoleApp with the corresponding sdk and run `dotnet build -v:diag` on each.
All SDKs were downloaded and installed via your `dotnet-install.sh` script.

3.0.100-preview4-010988: [diagPrev4.0.txt](https://github.com/dotnet/cli/files/3298747/diagPrev4.0.txt)

3.0.100-preview7-012430: [diagPrev7.0.txt](https://github.com/dotnet/cli/files/3298749/diagPrev7.0.txt)

If you want to try it by yourself, then don't use Windows 10 WSL Ubuntu 18.04 because it works with it. I'm using an actual Ubuntu 18.04 LTS Server and remoting via SSH.

**Comment by enricoreich** [comment]
Same problem on Debian 9.9 (stretch)
```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview7-012743
 Commit:    143bd43fcf

Runtime Environment:
 OS Name:     debian
 OS Version:  9
 OS Platform: Linux
 RID:         debian.9-x64
 Base Path:   /root/.dotnet/sdk/3.0.100-preview7-012743/

Host (useful for support):
  Version: 3.0.0-preview7-27826-20
  Commit:  ee0c7ead1a

.NET Core SDKs installed:
  3.0.100-preview7-012743 [/root/.dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0-preview7.19351.3 [/root/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0-preview7-27826-20 [/root/.dotnet/shared/Microsoft.NETCore.App]
```

**Comment by nkolev92** [comment]
Funny thing is we are now hitting the same issue on one of our build machines: 

https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2855648

2.2.302-preview-010246

It hangs in a very similar way.

//cc @donnie-msft has more info.

**Comment by nkolev92** [comment]
it only happens on 1 of our 4 linux machines. 
We could log into the machine and running any target on the offending projects hangs.

In our case running /t:clean hung the execution as well. 

Any ideas how to debug this? @peterhuene @livarcocc 
We have a machine that gets a 100% repro rate somehow.

**Comment by peterhuene** [comment]
One thing to quickly check if it's node reuse that is causing the hang by setting `MSBUILDDISABLENODEREUSE=1` in the environment prior to spawning `dotnet`.

Pinging @rainersigwald to see if he can help with any tips for aiding debugging of this issue.

If possible, I would use lldb + sos to dump the CLR threads from each MSBuild.dll dotnet process.

**Comment by rainersigwald** [comment]
I'd also be curious whether it works or produces a better error with `-m:1`.

**Comment by enricoreich** [comment]
Hello,

> One thing to quickly check if it's node reuse that is causing the hang by setting `MSBUILDDISABLENODEREUSE=1` in the environment prior to spawning `dotnet`.

Unfortunately the same problem: 
`dotnet build -v:diag > diag-3.0.100-preview8-013380.txt`: [diag-3.0.100-preview8-013380.txt](https://github.com/dotnet/cli/files/3418722/diag-3.0.100-preview8-013380.txt)


> I'd also be curious whether it works or produces a better error with -m:1.

`dotnet build -m:1 -v:diag > diag-3.0.100-preview8-013380_m1.txt`: [diag-3.0.100-preview8-013380_m1.txt](https://github.com/dotnet/cli/files/3418751/diag-3.0.100-preview8-013380_m1.txt)

Maybe it helps somehow.

**Comment by donnie-msft** [comment]
Our Build issues were resolved by setting `MSBUILDDISABLENODEREUSE=1` in [this PR](https://github.com/NuGet/NuGet.Client/pull/2956).

Thanks for the suggestion, and I've referenced an issue to revert this fix once this Issue is closed.

**Comment by lulzzz** [comment]
Same issue.

Using Azure DevOps Hosted ubuntu 16.04 fails to restore packages in private feeds.
dotnet core 3 preview 7
